### DISCUSSION TITLE
vector index

### DISCUSSION THREAD

**Discussion by mylee04** [microsoft/promptflow#2015] [General]
I made a vector index with my own data. But when I tried to use this index while interacting with chat model, the relevant source document was not provided, so I think I'm having trouble searching and extracting information from the indexed document. Where can I find this process in detail? 
<img width="707" alt="2024-02-13_16-23-15" src="https://github.com/microsoft/promptflow/assets/86863395/576e8eb8-d173-45d5-ad01-b178fca35cb1">

**Comment by brynn-code** [comment]
+ @dans-msft and @Adarsh-Ramanathan, could you please help on this? thanks.

**Reply by dans-msft** [reply]
The first step is to determine whether or not the relevant source document is being retrieved by the Index Lookup stage.  You can do this by looking at the output of the Index Lookup tool and seeing if your document is in there.  If it is not there then you have a retrieval issue -- something about your query relative to the way the data was indexed is causing your document not to be retrieved.  On the other hand, if the document *is* in the output of Index Lookup then you may be seeing a chat model hallucination.

Without any more information about the other steps in your flow it's hard to advise further.  Can you share more details about your flow plus information about how you built your index plus what sort of queries are producing this error?
